README.md

```diff
@@ -1,32 +1,88 @@
 
-> What if I have a question?
+- `presentation`: This layer contains all widgets for the app, responsible to display to the user the data coming from the `domain` and orchestrated by the application layer.
+- `application`: Orchestrates the communication from the `presentation` layer with the domain. No matter where the data originates, its **first destination** is going to be the application layer. TLDR: The role of the application layer is to decide "what to do next" with the data.
+- `domain`: Independent self-contained part of the application, not concerned with any technology used in the remaining of the app. Also responsible for determining how the other parts are structured and how they will interact with each other. Also containing the essential business logic. ( validation, data grouping, uniquely identify data)
+- `infrastructure`:  Deals with APIs, Firebase libraries, databases, and device sensors. Usually composed of data sources and data transfer objects.
 
-Just create a new issue in this repo and we will respond and get back to you quickly.
+## Features/comments
+
+### Linter
+For the linting rules we first enable all lint rules through the `all_lint_rules.yaml` file, and then through `analysis_options.yaml` we disable the ones that do not fit the current standard we are aiming for. This approach allows us to include the good practices AND the remaining rules in our coding style.
+
+  - It is easy to maintain our list when new rules are added. It is just a matter of copy-pasting the official list onto our all_lint_rules.yaml file.
+  - The selected set of rules were inspired by packages that are well estabilished flutter packages like: [provider](https://github.com/rrousselGit/provider) and [freezed](https://github.com/rrousselGit/freezed)
+  - This approach is opinionated and can be changed to adapt to the project needs.
+
+### Bloc (State Management)
+Bloc is a well-known and estabilished library to handle state management. It allows us to separate the presentation from the business logic improving the application scalability (especially used with a service locator pattern) and testability. The separation of the business logic (*application*) and the *presentation* makes this a very good approach to use with the Domain Driven Development (DDD)
+
+### Freezed package
+Provide us with helpful code generation that allows us to properly implement immutable objects around the project ( better conforming to the DDD approach). Also, help us with *classes* and *unions*.
+
+### Not using the `dartz` package
+Dartz package is an amazing tool with a lot of classes that allow us to better conform to the DDD approach, making error states 'unrepresentable'. Although for this challenge we will e using only a small subset of the entire library. That being said we manually implemented the class `Either` and `ValueObject` to use in this project.
+- Completely opinionated decisions. (avoid using extra libraries whenever possible)
+
```

Comment:
info: I really appreciate the call out to the fact that some things are opinionated decisions.  It is a subtle yet important part of understanding how to develop software as a part of a team.